v2/kafka-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlexIT.java

```diff
@@ -0,0 +1,234 @@
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineLauncher.LaunchInfo;
+import org.apache.beam.it.common.PipelineOperator.Result;
+import org.apache.beam.it.common.TestProperties;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.bigquery.BigQueryResourceManager;
+import org.apache.beam.it.gcp.bigquery.conditions.BigQueryRowsCheck;
+import org.apache.beam.it.gcp.bigtable.BigtableResourceManager;
+import org.apache.beam.it.kafka.KafkaResourceManager;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Integration test for {@link KafkaToBigQuery} (Kafka_To_BigQuery). */
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(KafkaToBigQueryFlex.class)
+@RunWith(JUnit4.class)
+public final class KafkaToBigQueryFlexIT extends TemplateTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigtableResourceManager.class);
+
+  private KafkaResourceManager kafkaResourceManager;
+  private BigQueryResourceManager bigQueryClient;
+
+  @Before
+  public void setup() throws IOException {
+    bigQueryClient = BigQueryResourceManager.builder(testName, PROJECT, credentials).build();
+    bigQueryClient.createDataset(REGION);
+
+    kafkaResourceManager =
+        KafkaResourceManager.builder(testName).setHost(TestProperties.hostIp()).build();
+  }
+
+  @After
+  public void tearDown() {
+    ResourceManagerUtils.cleanResources(kafkaResourceManager, bigQueryClient);
+  }
+
+  @Test
+  public void testKafkaToBigQuery() throws IOException {
+    baseKafkaToBigQuery(
+        b ->
+            b.addParameter("messageFormat", "JSON")
+                .addParameter("schemaFormat", "SINGLE_SCHEMA_FILE")
+                .addParameter("confluentAvroSchemaPath", "gs://FAKE")
+                .addParameter("schemaRegistryConnectionUrl", "")
+                .addParameter("binaryAvroSchemaPath", "gs://FAKE")
+                .addParameter("writeMode", "SINGLE_TABLE_NAME")
+                .addParameter("outputProject", "")
+                .addParameter("outputDataset", "")
+                .addParameter("bqTableNamePrefix", "")
+                .addParameter("useBigQueryDLQ", "false")
+                .addParameter("kafkaReadAuthenticationMode", "NONE")
+                .addParameter("kafkaReadUsernameSecretId", "")
+                .addParameter("kafkaReadPasswordSecretId", ""));
```

Comment:
we should remove these since these are optional. We should probably remove these from the other tests too but can be addressed in a separate PR. Can you add a TODO(to remove parameters that are not required) now if we are doing this in a different PR?

Reply:
Same for `schemaRegistryConnectionUrl`, `confluentAvroSchemaPath`, `schemaFormat`, `binaryAvroSchemaPath`. we should remove them from the test. I think having these here might hide errors we might get from the UI.

Reply:
Done